GeneratorInterface/GenFilters/src/LHEJetFilter.cc

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+
+#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/ClusterSequence.hh"
+#include "fastjet/Selector.hh"
+
+using namespace std;
+using namespace fastjet;
+
+class LHEJetFilter : public edm::EDFilter {
+public:
+  explicit LHEJetFilter(const edm::ParameterSet&);
+  ~LHEJetFilter() override {}
+
+private:
+  bool filter(edm::Event&, const edm::EventSetup&) override;
+
+  edm::EDGetTokenT<LHEEventProduct> tokenLHEEvent_;
+  double jetPtMin_;
+  JetDefinition jetdef_;
+
+  vector<PseudoJet> jetconsts_;
+};
+
+LHEJetFilter::LHEJetFilter(const edm::ParameterSet& params)
+    : tokenLHEEvent_(consumes<LHEEventProduct>(params.getParameter<edm::InputTag>("src"))),
+      jetPtMin_(params.getParameter<double>("jetPtMin")),
+      jetdef_(antikt_algorithm, params.getParameter<double>("jetR")) {}
+
+bool LHEJetFilter::filter(edm::Event& evt, const edm::EventSetup& params) {
+  edm::Handle<LHEEventProduct> lheinfo;
+  evt.getByToken(tokenLHEEvent_, lheinfo);
+
+  if (!lheinfo.isValid()) {
+    return true;
+  }
+
+  jetconsts_.clear();
```

Comment:
A local variable would be better than a member. A reasonable `reserve()` before the loop would be even better.